api/src/main/java/org/commonjava/indy/data/DefaultStoreValidator.java

```diff
@@ -0,0 +1,171 @@
+                RemoteRepository remoteRepository = (RemoteRepository) artifactStore;
+                boolean disabled = remoteRepository.isDisabled();
+                if(disabled) {
+                    throw new InvalidArtifactStoreException("Disabled Store", null, null);
+                }
+                //Validate URL from remote Repository URL
+                remoteUrl = Optional.of(new URL(remoteRepository.getUrl()));
+                // Check if remote.ssl.required is set to true and that remote repository protocol is https = throw IndyArtifactStoreException
+                if(configuration.isSSLRequired()
+                    && !remoteUrl.get().getProtocol().equalsIgnoreCase(StoreValidationConstants.HTTPS)) {
+
+                    //Check First if this remote repository is in allowed repositories from remote.nossl.hosts Config Variable
+                    List<String> remoteNoSSLHosts = configuration.getRemoteNoSSLHosts();
+                    String host = remoteUrl.get().getHost();
+
+                    for(String remoteHost : remoteNoSSLHosts) {
+                        LOGGER.warn(
+                            " RemoteHost " +host+ " in allowed list of remote hosts " + remoteNoSSLHosts + " is " +
+                                allowedNonSSLHostname(remoteHost, host) + " allowed " +
+                                " Remote URL: " + remoteUrl.get().toString()
+                        );
+                        // .apache.org , 10.192. , .maven.redhat.com
+                        if(allowedNonSSLHostname(remoteHost,host)) {
+                            errors.put(StoreValidationConstants.ALLOWED_SSL,remoteUrl.get().toString());
+                            LOGGER.warn(
+                                "NON-SSL RemoteRepository with URL: "+ host +" is ALLOWED under RULE: " + remoteHost
+                            );
+                        }
+                    }
+                }
+                // Execute HTTP GET & HEAD requests in separate thread pool from executor service
+                Future<Integer> httpGetStatus = executeGetHttp(new HttpGet(remoteUrl.get().toURI()), httpRequestsLatch);
+                Future<Integer> httpHeadStatus = executeHeadHttp(new HttpHead(remoteUrl.get().toURI()), httpRequestsLatch);
+                // Waiting for Http GET & HEAD Request Executor tasks to finish
+                httpRequestsLatch.await();
+                errors.put(StoreValidationConstants.HTTP_GET_STATUS, httpGetStatus.get().toString());
+                errors.put(StoreValidationConstants.HTTP_HEAD_STATUS, httpHeadStatus.get().toString());
+                if(!remoteUrl.get().getProtocol().equalsIgnoreCase(StoreValidationConstants.HTTPS)) {
+                    errors.put(StoreValidationConstants.HTTP_PROTOCOL, remoteUrl.get().getProtocol());
+                }
+                // Check for Sucessfull Validation
+                if (httpGetStatus.get() < 400 && httpHeadStatus.get() < 400) {
```

Comment:
If the protocol is not SSL, but GET and HEAD return 200, this will still return a valid result.

Reply:
This won't execute , because previously if it's not SSL then it will iterate trough all allowed hostnames or domains and return not valid or "SSL Allowed ...  by rule ... "

Reply:
Sorry, I don't see the return line you're mentioning here. Can you highlight it using a comment maybe? I guess we should have functional tests that prove this is working too...